sympy/stats/crv_types.py

```diff
@@ -2849,13 +2850,20 @@ class NormalDistribution(SingleContinuousDistribution):
 
     @staticmethod
     def check(mean, std):
-        _value_check(std > 0, "Standard deviation must be positive")
+        _value_check(std >= 0, "Standard deviation must be nonnegative.")
 
     def pdf(self, x):
+        # https://en.wikipedia.org/wiki/Normal_distribution#Zero-variance%20limit
+        if self.std == 0:
+            return DiracDelta(x - self.mean)
+
         return exp(-(x - self.mean)**2 / (2*self.std**2)) / (sqrt(2*pi)*self.std)
 
     def _cdf(self, x):
         mean, std = self.mean, self.std
+        if std == 0:
+            return Heaviside(x - mean)
```

Comment:
I think you want
```suggestion
            return Heaviside(x - mean, 1)
```
which should address your "probably more natural to make this 0" comment in the tests

Reply:
I completely missed that `H0` argument. Thanks. 

I was just about to implement it until I found a problem.
From the tests we have:
```python
X = Normal('X', 1, 0)
cdf(X)(1)  # 1
P(X <= 1)  # Heaviside(0)
```
 Even though `cdf(X)` uses the cdf directly,  `P(X <= 1)` uses the pdf and integrates it. I tried fixing it by setting `pdf = Heaviside(x - self.mean, H0=1).fdiff()` but it still seems to "forget" that `Heaviside(0) == 1` after integration. `DiracDelta()` did not have an `H0` argument.

Any suggestions on how to standardize the output?

Reply:
Sounds like you found a bug in `P()`.


I think for now it's probably ok to fix `cdf`, and leave `P(...)` as "broken"? It's not that the result of `P()` is incorrect, it's just not as useful as it could be - so could be patched in a more complex future PR.

Reply:
I think you're running into the same thing as https://github.com/sympy/sympy/issues/17383.

Reply:
All right. Updated the tests to match and to allow for `Heaviside(0)`. Too bad about the bug :(